src/sentry/eventstore/base.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import
+
+from sentry import nodestore
+from sentry.utils.services import Service
+
+
+class EventStorage(Service):
+    __all__ = (
+        'minimal_columns',
+        'full_columns',
+        'get_event_by_id',
+        'get_events',
+        'get_prev_event_id',
+        'get_next_event_id',
+        'bind_nodes',
+    )
+
+    # The minimal list of columns we need to get from snuba to bootstrap an
+    # event. If the client is planning on loading the entire event body from
+    # nodestore anyway, we may as well only fetch the minimum from snuba to
+    # avoid duplicated work.
+    minimal_columns = [
+        'event_id',
+        'group_id',
+        'project_id',
+        'timestamp',
+    ]
+
+    # A list of all useful columns we can get from snuba.
+    full_columns = minimal_columns + [
+        'culprit',
+        'location',
+        'message',
+        'platform',
+        'title',
+        'type',
+
+        # Required to provide snuba-only tags
+        'tags.key',
+        'tags.value',
+
+        # Required to provide snuba-only 'user' interface
+        'email',
+        'ip_address',
+        'user_id',
+        'username',
+    ]
+
+    def get_event_by_id(self, project_id, event_id, cols):
+        raise NotImplementedError
+
+    def get_events(self, start, end, cols, conditions, filter_keys, orderby, limit, offset):
+        raise NotImplementedError
+
+    def get_next_event_id(self, event, conditions, filter_keys):
```

Comment:
What's the difference between conditions and filter_keys ?

Reply:
AFAIK filter_keys will get translated into the `col IN list` syntax before being sent to Snuba https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L614-L620